bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/handler/SamsungTvHandler.java

```diff
@@ -81,301 +106,631 @@ public class SamsungTvHandler extends BaseThingHandler implements RegistryListen
+            // refresh channel as it's not polled
+            services.stream().filter(a -> a.getServiceName().equals(RemoteControllerService.SERVICE_NAME))
+                    .map(a -> a.handleCommand(channelUID.getId(), RefreshType.REFRESH));
+        }
+    }
 
-        for (SamsungTvService service : services) {
-            service.clearCache();
+    public void setModelName(String modelName) {
+        if (!modelName.isBlank()) {
+            this.modelName = modelName;
         }
     }
 
-    private synchronized void setPowerState(boolean state) {
+    public String getModelName() {
+        return modelName;
+    }
+
+    public synchronized void setPowerState(boolean state) {
         powerState = state;
     }
 
-    private synchronized boolean getPowerState() {
+    public synchronized boolean getPowerState() {
         return powerState;
     }
 
+    public synchronized boolean getArtModeSupported() {
+        return artModeSupported;
+    }
+
+    public synchronized void setArtModeSupported(boolean artmode) {
+        if (!artModeSupported && artmode) {
+            logger.debug("{}: ArtMode Enabled", host);
+        }
+        artModeSupported = artmode;
+    }
+
     @Override
     public void initialize() {
         updateStatus(ThingStatus.UNKNOWN);
 
-        logger.debug("Initializing Samsung TV handler for uid '{}'", getThing().getUID());
+        logger.debug("{}: Initializing Samsung TV handler for uid '{}'", host, getThing().getUID());
 
-        configuration = getConfigAs(SamsungTvConfiguration.class);
+        // note this can take up to 2 seconds to return if TV is off
+        discoverConfiguration();
 
         upnpService.getRegistry().addListener(this);
 
         checkAndCreateServices();
+    }
+
+    /**
+     * Start polling job with initial delay of 10 seconds if websocket protocol is selected
+     *
+     */
+    private void startPolling() {
+        int interval = configuration.getRefreshInterval();
+        int delay = configuration.isWebsocketProtocol() ? 10000 : 0;
+        pollingJob.ifPresentOrElse(job -> {
+            if (job.isCancelled()) {
+                pollingJob = Optional
+                        .of(scheduler.scheduleWithFixedDelay(this::poll, delay, interval, TimeUnit.MILLISECONDS));
+            } // else - scheduler is already running!
+        }, () -> {
+            logger.debug("{}: Start refresh task, interval={}", host, interval);
+            pollingJob = Optional
+                    .of(scheduler.scheduleWithFixedDelay(this::poll, delay, interval, TimeUnit.MILLISECONDS));
+        });
+    }
 
-        logger.debug("Start refresh task, interval={}", configuration.refreshInterval);
-        pollingJob = scheduler.scheduleWithFixedDelay(this::poll, 0, configuration.refreshInterval,
-                TimeUnit.MILLISECONDS);
+    private void stopPolling() {
+        pollingJob.ifPresent(job -> job.cancel(true));
```

Comment:
Because this way you can start and stop the polling job without getting a log entry every time you start it up again.

Reply:
It's not about logging. If you are not clearing the reference to the job after cancelling it, it will not be garbage collected, and you could even risk cancelling it one more time.